react/features/stream-effects/rnnoise/index.js

```diff
@@ -9,29 +9,17 @@ import RnnoiseProcessor from './RnnoiseProcessor';
 export { RNNOISE_SAMPLE_LENGTH } from './RnnoiseProcessor';
 export type { RnnoiseProcessor };
 
-let rnnoiseWasmInterface;
-let initializePromise;
+let rnnoiseModule;
 
 /**
  * Creates a new instance of RnnoiseProcessor.
  *
  * @returns {Promise<RnnoiseProcessor>}
  */
 export function createRnnoiseProcessor() {
-    if (!initializePromise) {
-        initializePromise = new Promise((resolve, reject) => {
-            rnnoiseWasmInterface = rnnoiseWasmInit({
-                onRuntimeInitialized() {
-                    resolve();
-                },
-                onAbort(reason) {
-                    reject(reason);
-                }
-            });
-        });
+    if (!rnnoiseModule) {
+        rnnoiseModule = rnnoiseWasmInit();
```

Comment:
I remembered having some issues with the rnnoise promise based init (that's why all the nice boilerplate). I assume you tested with an image build using a newer version of emscripten.
This really looks much better.

Reply:
Indeed!